### PR TITLE
[docs] refactor forms docs

### PR DESCRIPTION
**Pull request by giladgray**
a la #2574

### CODE REVIEW

File: packages/core/src/components/forms/control-group.md [hunk 1]
```diff
@@ -1,50 +1,43 @@
...
 
 <div class="@ns-callout @ns-intent-success @ns-icon-comparison">
     <h4 class="@ns-heading">Control group vs. input group</h4>
     <p>Both components group multiple elements into a single unit, but their usage patterns are
-    different.</p>
-    <p>Think of `.@ns-control-group` as a parent with multiple children, each of them a
-    "control."</p>
-    <p>Conversely, a `.@ns-input-group` is a single control, and should function like so. A
+    quite different.</p>
+    <p>Think of `ControlGroup` as a parent with multiple children, each of them a separate
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: The wording of "each of them a separate control" feels a bit awkward. I'd recommend either:
"with each of them..." or
"with each one..."

File: packages/core/src/components/forms/control-group.md [hunk 2]
```diff
@@ -1,50 +1,43 @@
...
-@css control-group-fill
+@## Flex layout
 
-@## Vertical control groups
+`ControlGroup` is a CSS inline flex row (or column if vertical) and provides
+some modifer props for common flexbox patterns:
 
-Add the class `@ns-vertical` to create a vertical control group. Controls in a vertical group
-will all have the same width as the widest control.
+- Enable the `fill` prop on a control group to make all controls expand equally to
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: prop -> property

File: packages/core/src/components/forms/control-group.md [hunk 3]
```diff
@@ -1,50 +1,43 @@
...
+  fill the available space.
+    - Controls will expand horizontally by default, or vertically if the `vertical` prop is enabled.
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: prop -> property

File: packages/core/src/components/forms/control-group.md [hunk 4]
```diff
@@ -1,50 +1,43 @@
...
+    - Add the class `Classes.FIXED` to individual controls to revert them to their initial sizes.
 
-@css control-group-vertical
+- Alternatively, enable the `fill` prop on specific controls (instead of on the
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: prop -> property

File: packages/core/src/components/forms/control-group.md [hunk 5]
```diff
@@ -1,50 +1,43 @@
...
+  group) to expand them equally to fill the available space while other
+  controls retain their original sizes.
 
-@## JavaScript API
+You can adjust the specific size of a control with the `flex-basis` or `width`
+CSS properties.
 
-The `ControlGroup` component is available in the **@blueprintjs/core** package. Make sure to review [getting started docs for installation info](#blueprint/getting-started).
+@## Props
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: prop -> property

File: packages/core/src/components/forms/control-group.md [hunk 6]
```diff
@@ -1,50 +1,43 @@
...
 
-This component is a simple wrapper around the corresponding CSS API. It supports the full range of HTML props.
+This component is a lightweight wrapper around its children. It supports all
+HTML `<div>` props, in addition to those listed below.
```
Comment (benjdj6) [inline packages/core/src/components/forms/control-group.md]: prop -> property

File: packages/core/src/components/forms/form-group.md [hunk 7]
```diff
@@ -1,31 +1,15 @@
...
+@# Form group
 
 Form groups support more complex form controls than [simple labels](#core/components/forms/label.simple-labels),
 such as [control groups](#core/components/forms/control-group) or [`NumericInput`](#core/components/forms/numeric-input).
 They also support additional helper text to aid with user navigation.
 
 @reactExample FormGroupExample
 
-@## CSS API
+@## Props
```
Comment (benjdj6) [inline packages/core/src/components/forms/form-group.md]: prop -> property

File: packages/core/src/components/forms/form-group.md [hunk 8]
```diff
@@ -1,31 +1,15 @@
...
-
-@css form-group
-
-@## JavaScript API
-
-The `FormGroup` component is available in the __@blueprintjs/core__ package.
-Make sure to review the [getting started docs for installation info](#blueprint/getting-started).
-
-This component is a simple wrapper around the CSS API that abstracts away the HTML complexity.
+This component is a lightweight wrapper around its children with props for the
```
Comment (benjdj6) [inline packages/core/src/components/forms/form-group.md]: prop -> property

File: packages/core/src/components/forms/text-inputs.md [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+
+@## Input group
+
+An input group allows you to add icons and buttons _within_ a text input to expand its
+functionality. For example, you might use an input group to build a visibility toggle for a password
+field.
+
+@reactExample InputGroupExample
+
+@### Props
```
Comment (benjdj6) [inline packages/core/src/components/forms/text-inputs.md]: prop -> property

File: packages/core/src/components/forms/text-inputs.md [hunk 10]
```diff
@@ -0,0 +1,85 @@
+@# Text inputs
+
+Blueprint provides two ways to create a text input:
+
+1. CSS only: apply `Classes.INPUT` to an `<input>` or `<textarea>` element to
+   style the native HTML tag.
+1. React component: use the `InputGroup` component for an advanced single-line
```
Comment (benjdj6) [inline packages/core/src/components/forms/text-inputs.md]: This should be "2."

File: packages/core/src/components/forms/text-inputs.md [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+@### Props
+
+The `InputGroup` React component  supports one non-interactive icon on the left
+side and one arbitrary element on the right side. Unlike the CSS approach,
+`InputGroup` supports _content of any length_ on the right side (not just
+icon buttons) because it is able to measure the content and ensure there is
+always space for it.
+
+`InputGroup` can be used just like a standard React `input` element, in
+controlled or uncontrolled fashion. In addition to its own props, it supports
```
Comment (benjdj6) [inline packages/core/src/components/forms/text-inputs.md]: "in **a** controlled or uncontrolled..."
prop -> property

File: packages/core/src/components/forms/text-inputs.md [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+all valid props for HTML `<input>` elements and proxies them to that element in
```
Comment (benjdj6) [inline packages/core/src/components/forms/text-inputs.md]: prop -> property

File: packages/core/src/components/forms/text-inputs.md (hunk 10 above)
Comment (giladgray) [reply packages/core/src/components/forms/text-inputs.md]: markdown auto-increments when rendering. i prefer making them all the same number so it's easier to edit later.

**Pull request merged**